docs/source/tutorial/convnet.rst

```diff
@@ -0,0 +1,428 @@
+            w = chainer.initializers.HeNormal()
+            super(BottleNeck, self).__init__(
+                conv1x1a=L.Convolution2D(
+                    n_in, n_mid, 1, stride, 0, initialW=w, nobias=True),
+                conv3x3b=L.Convolution2D(
+                    n_mid, n_mid, 3, 1, 1, initialW=w, nobias=True),
+                conv1x1c=L.Convolution2D(
+                    n_mid, n_out, 1, 1, 0, initialW=w, nobias=True),
+                bn_a=L.BatchNormalization(n_mid),
+                bn_b=L.BatchNormalization(n_mid),
+                bn_c=L.BatchNormalization(n_out))
+            if proj:
+                self.add_link('conv1x1r', L.Convolution2D(
+                    n_in, n_out, 1, stride, 0, initialW=w, nobias=True))
+                self.add_link('bn_r', L.BatchNormalization(n_out))
+            self.proj = proj
+
+        def __call__(self, x, train):
+            h = F.relu(self.bn_a(self.conv1x1a(x), test=not train))
+            h = F.relu(self.bn_b(self.conv3x3b(x), test=not train))
+            h = self.bn_c(self.conv1x1c(x), test=not train)
+            if self.proj:
+                x = self.bn_r(self.conv1x1r(x), test=not train)
+            return F.relu(h + x)
+
+In :class:`BottleNeck` class, it switches whther ``conv1x1r`` to fit the
+channel dimension of the input ``x`` and the output ``h`` and ``bn_r`` for it
+should be added or not with respect to :attr:`~BottleNeck.proj` attribute.
+Writing the building block in this way gains the re-usability of a class a lot.
+It switches not only the behavior in :meth:`__class__` by flags but also the
+parameter registration. In this case, when :attr:`proj` is ``False``, the
+:class:`BottleNeck` doesn't have `conv1x1r` and `bn_r` layers, so the memory
+usage would be efficient compared to the case when it registers both anyway and
+just ignore them if :attr:`proj` is ``False``.
+
+Using nesting :class:`~chainer.Chain` s and :class:`~chainer.ChainList` for
+sequential part enables us to write complex and very deep models easily.
+
+Use Pre-trained Models
+''''''''''''''''''''''
+
+Various ways to write your models is described above. And VGG16 and ResNet are
+very useful as general feature extractor for many different tasks from image
+classification. So Chainer provides you the pre-trained models of VGG16 and
+ResNet-50/101/152 models with a simple API. You can use those models in this
+way:
+
+.. doctest::
+
+    from chainer.links import VGG16Layers
+
+    model = VGG16Layers()
+
+When :class:`~chainer.links.VGG16Layers` is instantiated, the pre-trained
+parameters is automatically downloaded from the author's server. So you can
+immediately start to use VGG16 with pre-trained weight as a good image feature
+extractor. See the details of this model here:
+:class:`chainer.links.VGG16Layers`.
+
+In the case of ResNet models, there are 3 variation in terms of the number of
+layers. We have :class:`chainer.links.ResNet50`,
```

Comment:
In the case of ResNet models, there are three variations differing in the number of layers.